hardware/opas/TOPAS/TOPAS.py

```diff
@@ -36,13 +36,29 @@ def __init__(self, *args, **kwargs):
         self.curve_paths = collections.OrderedDict()
         self.ini = project.ini_handler.Ini(os.path.join(main_dir, 'hardware', 'opas', 'TOPAS', 'TOPAS.ini'))
         self.has_shutter = kwargs['has_shutter']
+        self.yaq_port = kwargs['yaq_port']
         if self.has_shutter:
             self.shutter_position = pc.Bool(name='Shutter', display=True, set_method='set_shutter')
         BaseDriver.__init__(self, *args, **kwargs)  
+        self.serial_number = self.ini.read('OPA' + str(self.index), 'serial number')
+        # load api
+        self.api = yaqc.Client(self.yaq_port)
+        if self.has_shutter:
+            self.api.set_shutter(False)
+        
+        # motor positions
+        for motor_name in self.motor_names:
+            min_position, max_position = self.api.get_motor_range(motor_name)
+            limits = pc.NumberLimits(min_position, max_position)
+            number = pc.Number(initial_value=0, limits=limits, display=True, decimals=6)
+            self.motor_positions[motor_name] = number
+            self.recorded['w%d_'%self.index + motor_name] = [number, None, 1., motor_name]
```

Comment:
This block simply moved from the `initialize` function (no longer overridden)

Nothing here is particularly computationally intensive, and the UI was failing to be built properly if the motor dicts and yaq client were not there at `__init__` time... Not sure why that changed to be honest, but moving this here fixed that...

Because the range is retrieved from the api call, the yaq client loading is also done in `__init__`